Automattic/wp-calypso
# Plans: Add tiered Backup products to the list of features indicating a site has Backup

#### Changes proposed in this Pull Request

These changes deal specifically with the exit survey that appears when people decide to cancel a Jetpack product. On this form we check to see which features the customer wants to cancel, so that we can do a better job to retain them.

* Add new tier 1/2 Backup products to the list of checked features when determining `productHasBackup`.
* Create and expose a `isJetpackSearchSlug` convenience function in `@automattic/calypso-products`, since similar functions exist for all other products.
* Refactor all `feature-checks` methods to use `planHasAtLeastOneFeature` instead of repeated OR conditions.

#### Testing instructions

* Verify all unit tests pass, especially `client/blocks/jetpack-benefits/test/feature-checks.js`.
* If you'd like, you can check to see that Backup shows up on the list of benefits when you try to cancel a purchase of a tiered Backup product (e.g., Jetpack Backup (20 GB), Jetpack Security (1 TB)). To do this, purchase a relevant product and then go to **Upgrades > Purchases** in Calypso Blue and click **Remove <ProductName>**. You should see a cancellation survey similar to the below reference screenshot if Backup is a part of the product you purchased.

#### Reference screenshots

##### Backup included

<img width="1218" alt="image" src="https://user-images.githubusercontent.com/670067/136438405-31e9ed57-7e58-4575-bb9e-32505768c8dd.png">

##### Backup not included

<img width="1194" alt="image" src="https://user-images.githubusercontent.com/670067/136439814-7b1a9929-db4a-4d85-868e-5fe7f42f2296.png">
